### ISSUE TITLE
Broken Images in Posts Due to Incorrect Hostname

### ISSUE THREAD

**Issue by syin** [bitnami/bitnami-docker-discourse#32]
I am having issues with the discourse hostname when images are uploaded in a post. When a post is first created, the image url used is a relative url (`/images/....`), which works, but after a page reload the image url changes to hardcode `www.example.com` as the host (`//www.example.com/images/`) instead of the actual domain we are using. 

For example, this is how an emoji is displaying:
```
<img src="//www.example.com/images/emoji/emoji_one/open_mouth.png?v=3" title=":open_mouth:" class="emoji" alt=":open_mouth:">
```

Things I have tried (rebuilding the containers and rebaking posts after each):
- setting the environment variable `DISCOURSE_HOSTNAME='mysite.com'`. I am not sure if this is relevant to these docker containers, since the docker containers aren't using an `app.yml` like the official Discourse install is. (See https://meta.discourse.org/t/main-domain-used-as-hostname-instead-of-discourse-hostname-for-emoji-path/69979)
- editing `/bitnami/discourse/conf/database.yml`
```
production:
...
  host_names:
    ### Don't include the port number here. Change the "port" site setting instead, at /admin/site_settings.
    ### If you change this setting you will need to
    ###   - restart sidekiq if you change this setting
    ###   - rebake all to posts using: `RAILS_ENV=production bundle exec rake posts:rebake`
    - "mysite.com"
```
- setting the hostname in `/bitnami/discourse/conf/discourse.conf`. I am not sure if this is being used either?
```
hostname = "mysite.com"
```

Where should I be setting the discourse hostname? I couldn't find any information about it in the docs or from google searching. Thanks!

## Additional info:

- Host OS and version: Ubuntu 16.04, hosted on EC2
- Output of `docker info`:
```
Containers: 4
 Running: 4
 Paused: 0
 Stopped: 0
Images: 50
Server Version: 1.13.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 150
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 03e5862ec0d8d3b3f750e19fca3ee367e13c090e
runc version: 2f7393a47307a16f8cee44a37b262e8b81021e3e
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-97-generic
Operating System: Ubuntu 16.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 7.795 GiB
Name: ip-172-31-31-110
ID: 6TAI:UXJN:KLWG:EEV4:JOIE:C6AH:GFEO:FCBN:TKEP:BM6U:QPPY:KIDI
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```
- Output of `$BITNAMI_IMAGE_VERSION` (there is no `$BITNAMI_APP_VERSION` variable): 1.8.9-r1
- Output of `docker version`:
```
Client:
 Version:      1.13.0
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Tue Jan 17 09:58:26 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.0
 API version:  1.25 (minimum version 1.12)
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Tue Jan 17 09:58:26 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Comment by juan131** [comment]
On the one hand, at this moment, it's not possible to configure the Discourse Hostname using environment variables the same way you can configure, for instance, the Password using `DISCOURSE_PASSWORD`. @tompizmor I think it could be interesting to add this new parameter, what do you think?

Regarding how to manually configure this setting, it should be enough by adding the line below to `/bitnami/discourse/conf/discourse.conf`:

```
  hostname = my-site.com
```

And `/bitnami/discourse/conf/database.yml` adding the line below under "production" -> "host_names" section:

```
   - "my-site.com"
```

Then restart Discourse: `nami restart discourse`

**Comment by clkao** [comment]
apparently you need to do the same change @juan131 suggested in the discourse-sidekiq container too, and run `discourse-sidekiq` then `posts:rebake`

**Comment by syin** [comment]
@rene-gomez I experienced the same thing and I ended up using the official discourse docker containers which worked :neutral_face:

**Comment by juan131** [comment]
Hi @rene-gomez 

Did you [persist your Discourse data](https://github.com/bitnami/bitnami-docker-discourse#persisting-your-application) when you created your containers? Any other way the configuration data will be lost when you stop your containers.

**Comment by beltran-rubo** [comment]
Hi,

We released a new revision "2.0.3-r11" where the domain in the email notifications is configurable using the ENV var `DISCOURSE_HOSTNAME`. The default value is `www.example.com`.
